python-spec/src/somacore/data.py

```diff
@@ -45,6 +45,7 @@ def create(
         *,
         schema: pa.Schema,
         index_column_names: Sequence[str] = (options.SOMA_JOINID,),
+        domains: Optional[Sequence[Optional[Tuple[Any, Any]]]] = None,
```

Comment:
the thing missing from this is access to the domain/extent on an existing DataFrame.

We used to have `shape` (it still exists on NDArray), and I thought we intended to bring that back once the DataFrame had added this first-class concept.

So, where is the `shape` property equivalent?   And what should it be called?   (`domain_extent`?)

CC: @thetorpedodog